services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/grpc/LogicalPortSpec.groovy

```diff
@@ -0,0 +1,78 @@
+package org.openkilda.functionaltests.spec.grpc
+
+
+import static org.junit.Assert.assertTrue
+import static org.openkilda.testing.ConstantsGrpc.LOGICAL_PORT
+import static org.openkilda.testing.ConstantsGrpc.SWITCH
+import static org.openkilda.testing.ConstantsGrpc.SWITCH_PORT
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.grpc.speaker.model.LogicalPortDto
+import org.openkilda.messaging.error.MessageError
+
+import groovy.util.logging.Slf4j
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Unroll
+
+@Slf4j
+@Narrative("Test Logical Port.")
+class LogicalPortSpec extends BaseSpecification {
+    @Unroll
+    def "Not able to create logical port with incorrect ports number(lPort/sPort): #logicalPortNumber/#portNumber."() {
+        when:
+        " Try to create with incorrect values (logical port/simple port): $logicalPortNumber/$portNumber."
+        grpc.createLogicalPort(SWITCH, new LogicalPortDto([portNumber], logicalPortNumber))
+
+        then: "Human readable error is returned."
+        def exc = thrown(HttpClientErrorException)
+        assert exc.statusCode == HttpStatus.BAD_REQUEST
+        assert exc.responseBodyAsString.to(MessageError).errorMessage == errorMessage
+
+        where:
+        logicalPortNumber | portNumber  | errorMessage
+        99                | SWITCH_PORT | "Valid logicalportno range is 100 to 63487."
```

Comment:
`Valid logicalportno range is 100 to 63487` these values are unique for each switch. You can't assume this error message for any switch